algorithmic_efficiency/workloads/imagenet_resnet/imagenet_v2.py

```diff
@@ -0,0 +1,62 @@
+"""ImageNet-v2 tf.data input pipeline.
+
+Uses TFDS https://www.tensorflow.org/datasets/catalog/imagenet_v2.
+"""
+import functools
+
+import jax
+import numpy as np
+import tensorflow_datasets as tfds
+
+from algorithmic_efficiency.workloads.imagenet_resnet.imagenet_jax import \
+    input_pipeline
+
+
+def _shard(x):
```

Comment:
`algorithmic_efficiency.data_utils.shard_numpy_ds` could be adjusted to be also used here.

Reply:
I was going to use that but we convert these to numpy first so the `x = x._numpy()` code will fail in there. But we absolutely need to clean up how we do tf.tensor/np.ndarray/torch.tensor/jax.devicearray handling/padding/conversion with datasets, I plan on doing that once the fires have been put out with implementing them all

Reply:
Yeah it would need to be adjusted first. And yes, we'll have to clean up related stuff anyway ;)